src/App.tsx

```diff
@@ -1,19 +1,138 @@
 
+import todos from './api/todos';
 import users from './api/users';
 
-const App: React.FC = () => {
+import { User } from './components/UserInfo/UserInfo';
+import { Todo } from './components/TodoInfo/TodoInfo';
+import { TodoList } from './components/TodoList/TodoList';
+
+export interface PreparedTodo extends Todo {
+  user: User | null;
+}
+
+export const preparedTodos: PreparedTodo[] = todos.map((todo) => ({
+  ...todo,
+  user: users.find((user) => (
+    user.id === todo.userId
+  )) || null,
+}) as PreparedTodo);
+
+export const App: React.FC = () => {
+  const [todo, setTodos] = useState([...preparedTodos]);
+  const [title, setTitle] = useState('');
+  const [titleDirty, setTitleDirty] = useState(false);
+  const [userName, setUserName] = useState('Choose a user');
+  const [userNameDirty, setUserNameDirty] = useState(false);
+
+  const handleInputChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setTitle(event.target.value);
+    setTitleDirty(false);
+  };
+
+  const handleSelectChange = (event: React.ChangeEvent<HTMLSelectElement>) => {
+    const { value } = event.target;
+
+    setUserName(value);
+    setUserNameDirty(false);
+  };
+
+  const addTodo = () => {
+    const newUser = users.find(
+      user => user.name === userName,
+    ) || null;
+
+    let userId = 0;
+
+    if (newUser) {
+      userId = newUser.id;
+    }
+
+    const newTodo = {
+      id: Math.max(...todos.map(t => t.id)) + 1,
+      title,
+      userId,
+      completed: false,
+      user: newUser,
+    };
+
+    if (newTodo.title.length === 0) {
+      return;
+    }
+
+    if (!newTodo.user?.name) {
+      return;
+    }
+
+    setTodos([...todo, newTodo]);
+  };
+
+  const handleFormSubmit = (event: React.ChangeEvent<HTMLFormElement>) => {
+    event.preventDefault();
+
+    if (!title || !userName) {
+      setUserNameDirty(!userName);
+      setTitleDirty(!title);
+    }
+
+    addTodo();
+
+    setTitle('');
+    setUserName('');
+  };
+
   return (
     <div className="App">
-      <h1>Add todo form</h1>
+      <div className="App__container">
+        <form
+          onSubmit={handleFormSubmit}
+          className="App__form"
+        >
+          {titleDirty && <div className="App__form-error">Ведіть Title</div>}
+          <input
+            className="App__input"
+            type="text"
+            placeholder="write title"
+            name="title"
+            value={title}
+            onChange={handleInputChange}
+          />
+          {userNameDirty && <div className="App__form-error">Error</div>}
+          <select
+            className="App__select"
+            data-cy="userSelect"
+            name="name"
+            value={userName}
+            onChange={handleSelectChange}
+          >
+            <option className="App__option">
+              Choose a user
+            </option>
+            {
+              users.map((user) => (
+                <option
+                  key={user.id}
+                >
```

Comment:
you can fit 1 or 2 attributes in one line easily, fix it everywhere you have it
```suggestion
                <option key={user.id} >
```